models/yang/sonic/sonic-routing-policy-sets.yang

```diff
@@ -0,0 +1,175 @@
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC ROUTING POLICY SETS";
+
+    revision 2019-10-23 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-routing-policy-sets {
+
+        container PREFIX_SET {
+
+            list PREFIX_SET_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,71})';
+                        length 1..72;
+                    }
+                }
+
+                leaf mode {
+                    type enumeration {
+                        enum IPv4;
+                        enum IPv6;
+                    }
+                }
+            }
+        }
+
+        container PREFIX {
+
+            list PREFIX_LIST { 
+                key "set_name ip_prefix masklength_range";
+
+                leaf set_name {
+                    mandatory true;
+                    type leafref {
+                        path "../../../PREFIX_SET/PREFIX_SET_LIST/name";
+                    }
+                }   
+
+                leaf ip_prefix {
+                    /* TODO - Validate ip-prefix based on PREFIX_SET/mode */
+                    type inet:ip-prefix;
+                    mandatory true;
+                    description
+                        "The prefix member in CIDR notation -- while the
+                        prefix may be either IPv4 or IPv6, most
+                        implementations require all members of the prefix set
+                        to be the same address family.  Mixing address types in
+                        the same prefix set is likely to cause an error.";
+                }
+
+                leaf masklength_range {
+                    type string;
+                    mandatory true;
+                    description
+                        "Defines a range for the masklength, or 'exact' if
+                        the prefix has an exact length.
+
+                        Example: 10.3.192.0/21 through 10.3.192.0/24 would be
+                        expressed as prefix: 10.3.192.0/21,
+                                   masklength_range: 21..24.
+
+                        Example: 10.3.192.0/21 would be expressed as
+                                  prefix: 10.3.192.0/21,
+                                  masklength_range: exact";
+                }
+            }
+        }
+        container COMMUNITY_SET {
+
+            list COMMUNITY_SET_LIST {
+                key "community_set_name";
+
+                leaf community_set_name {
+                    mandatory true;
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,71})';
+                        length 1..72;
+                    }
+                }
+
+                leaf type {
```

Comment:
Not sure if we can avoid this name "type". "type" is a reserved YANG keyword.

Reply:
changed to set_type